src/main/java/com/powsybl/openloadflow/ac/equations/ClosedBranchSide1IntensityMagnitudeEquationTerm.java

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright (c) 2019, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.openloadflow.ac.equations;
+
+import com.powsybl.openloadflow.equations.Variable;
+import com.powsybl.openloadflow.equations.VariableSet;
+import com.powsybl.openloadflow.network.LfBranch;
+import com.powsybl.openloadflow.network.LfBus;
+import net.jafama.FastMath;
+
+import java.util.Objects;
+
+import static com.powsybl.openloadflow.network.PiModel.A2;
+import static com.powsybl.openloadflow.network.PiModel.R2;
+
+/**
+ * @author Gael Macherel <gael.macherel at artelys.com>
+ */
+public class ClosedBranchSide1IntensityMagnitudeEquationTerm extends AbstractClosedBranchAcFlowEquationTerm {
+
+    private double i1;
+
+    private double di1dv1;
+
+    private double di1dv2;
+
+    private double di1dph1;
+
+    private double di1dph2;
+
+    private double di1da1;
+
+    public ClosedBranchSide1IntensityMagnitudeEquationTerm(LfBranch branch, LfBus bus1, LfBus bus2, VariableSet variableSet,
+                                                           boolean deriveA1, boolean deriveR1) {
+        super(branch, bus1, bus2, variableSet, deriveA1, deriveR1);
+    }
+
+    @Override
+    protected double calculateDer(double dph1, double dph2, double dv1, double dv2, double a1, double r1) {
+        return di1dph1 * dph1 + di1dph2 * dph2 + di1dv1 * dv1 + di1dv2 * dv2;
+    }
+
+    @Override
+    public void update(double[] x) {
+        Objects.requireNonNull(x);
+        double v1 = x[v1Var.getRow()];
+        double v2 = x[v2Var.getRow()];
+        double ph1 = x[ph1Var.getRow()];
+        double ph2 = x[ph2Var.getRow()];
+        double theta = ksi - (a1Var != null && a1Var.isActive() ? x[a1Var.getRow()] : branch.getPiModel().getA1())
+                + A2 - ph1 + ph2;
+        double sinTheta = FastMath.sin(theta);
+        double cosTheta = FastMath.cos(theta);
+        double r1 = r1Var != null && r1Var.isActive() ? x[r1Var.getRow()] : branch.getPiModel().getR1();
+        double p1 = r1 * v1 * (g1 * r1 * v1 + y * r1 * v1 * sinKsi - y * R2 * v2 * sinTheta);
+        double q1 = r1 * v1 * (-b1 * r1 * v1 + y * r1 * v1 * cosKsi - y * R2 * v2 * cosTheta);
+
+        i1 = Math.hypot(p1, q1) / (Math.sqrt(3.) * v1 / 1000);
```

Comment:
@Hadrien-Godard @Djazouli I think we don't have to compute the intensity from p and q. I have done that before as it was a post-processing. Can we compute the intensity through the Ohm law ?

Reply:
Yes we can, but we will end up with the same formula since Ohm law is used to obtain above expression of p and q from v and phi.
The difficulty here is that Ohm law expresses a difference of complex tensions in term of a complex intensity, or the intensity that is measured (here i1) is in fact the magnitude ofthe complex intensity. Therefore the Math.hypot is needed to express this magnitude.

Reply:
Yes but it would be simpler to calculate re(i) and img(i) and then mod(i) that calculating p and q and then dividing by voltage for nothing

Reply:
Sure it would be more efficient.